app/api/Torrent.js

```diff
@@ -48,8 +48,21 @@ export default class Torrent {
 
     this.engine.add(magnetURI, { path: cacheLocation }, torrent => {
       const server = torrent.createServer();
-      server.listen(port);
-      this.server = server;
+      const thisRef = this;
+
+      detect(this.port).then((_port) => {
```

Comment:
And yeah, if you notice, I added a new member `port` to the object. I did it because the already present `port` variable was declared as a `const`. So, I couldn't assign it different values when the port was already occupied, so I created a new member and gave it a default value 9090.